service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/FurtherEvidenceUploadedEventHandler.java

```diff
@@ -32,34 +38,76 @@ public void handleDocumentUploadedEvent(final FurtherEvidenceUploadedEvent event
         final CaseData caseDataBefore = event.getCaseDataBefore();
         final UserDetails uploader = event.getInitiatedBy();
 
+        DocumentUploadNotificationUserType userType = event.getUserType();
+        List<String> newNonConfidentialDocuments = getNewNonConfidentialBundleNames(caseData, caseDataBefore, userType);
+
         final Set<String> recipients = new HashSet<>();
 
-        if (event.isUploadedByLA()) {
-            if (hasNewNonConfidentialDocuments(caseData.getFurtherEvidenceDocumentsLA(),
-                caseDataBefore.getFurtherEvidenceDocumentsLA())) {
-                recipients.addAll(furtherEvidenceNotificationService.getRepresentativeEmails(caseData));
-            }
-        } else {
-            if (hasNewNonConfidentialDocuments(caseData.getFurtherEvidenceDocuments(),
-                caseDataBefore.getFurtherEvidenceDocuments())) {
-                recipients.addAll(furtherEvidenceNotificationService.getRepresentativeEmails(caseData));
+        if (!newNonConfidentialDocuments.isEmpty()) {
+            recipients.addAll(furtherEvidenceNotificationService.getRepresentativeEmails(caseData));
+
+            if (userType != LOCAL_AUTHORITY) {
                 recipients.addAll(furtherEvidenceNotificationService.getLocalAuthoritySolicitorEmails(caseData));
             }
         }
 
         recipients.removeIf(email -> Objects.equals(email, uploader.getEmail()));
 
         if (isNotEmpty(recipients)) {
-            furtherEvidenceNotificationService.sendNotification(caseData, recipients, uploader.getFullName());
+            furtherEvidenceNotificationService.sendNotification(caseData, recipients, uploader.getFullName(),
+                newNonConfidentialDocuments);
+        }
+    }
+
+
+    private List<String> getNewNonConfidentialBundleNames(CaseData caseData, CaseData caseDataBefore,
+                                                          DocumentUploadNotificationUserType userType) {
+
+        List<Element<SupportingEvidenceBundle>> newBundle = getEvidenceBundle(caseData, userType);
+        List<Element<SupportingEvidenceBundle>> oldBundle = getEvidenceBundle(caseDataBefore, userType);
+
+        List<String> documentNames = new ArrayList<String>();
+
+        unwrapElements(newBundle).forEach(newDoc -> {
+            if (!newDoc.isConfidentialDocument() && !unwrapElements(oldBundle).contains(newDoc)) {
+                documentNames.add(newDoc.getName());
+            }
+        });
+        return documentNames;
+    }
+
+    private List<Element<SupportingEvidenceBundle>> getEvidenceBundle(CaseData caseData,
+                                                                      DocumentUploadNotificationUserType userType) {
+        if (userType == LOCAL_AUTHORITY) {
+            return caseData.getFurtherEvidenceDocumentsLA();
+        } else if (userType == SOLICITOR) {
+            List<Element<SupportingEvidenceBundle>> furtherEvidenceBundle =
```

Comment:
if you are annoyed by the long line length then you can use the `var` keyword if you want instead of defining it as `List<Element<SupportingEvidenceBundle>>`